src/mlpack/bindings/python/print_input_processing.hpp

```diff
@@ -122,21 +122,48 @@ void PrintInputProcessing(
   std::cout << prefix << "# Detect if the parameter was passed; set if so."
       << std::endl;
   if (!d.required)
-  {
-    std::cout << prefix << "if " << d.name << " is not None:" << std::endl;
+  { 
+    if (GetArmaType<T>() == "row" || GetArmaType<T>() == "col")
+    {
+      std::cout << prefix << "if " << d.name << " is not None:" << std::endl;
 
-    std::cout << prefix << "  " << d.name << "_tuple = to_matrix(" << d.name
-        << ", dtype=" << GetNumpyType<typename T::elem_type>() << ", "
-        << "copy=CLI.HasParam('copy_all_inputs'))" << std::endl;
-    std::cout << prefix << "  " << d.name << "_mat = arma_numpy.numpy_to_"
-        << GetArmaType<T>() << "_" << GetNumpyTypeChar<T>() << "(" << d.name
-        << "_tuple[0], " << d.name << "_tuple[1])" << std::endl;
-    std::cout << prefix << "  SetParam[" << GetCythonType<T>(d) << "](<const "
-        << "string> '" << d.name << "', dereference(" << d.name << "_mat))"
-        << std::endl;
-    std::cout << prefix << "  CLI.SetPassed(<const string> '" << d.name << "')"
-        << std::endl;
-    std::cout << prefix << "  del " << d.name << "_mat";
+      std::cout << prefix << "  " << d.name << "_tuple = to_matrix(" << d.name
+          << ", dtype=" << GetNumpyType<typename T::elem_type>() << ", "
+          << "copy=CLI.HasParam('copy_all_inputs'))" << std::endl;
+      std::cout << prefix << "  " << "if len(" << d.name << "_tuple[0].shape"
+	  << ") > 1:" << std::endl;
+      std::cout << prefix << "  " << prefix << "if " << d.name << "_tuple[0]"
+	  << ".shape[0] == 1 or " << d.name << "_tuple[0].shape[1] == 1:" 
+	  << std::endl;
+      std::cout << prefix << "  " << prefix << "  " << d.name << "_tuple[0] = " 
+	  << d.name << "_tuple" << "[0].ravel()" << std::endl;
```

Comment:
I think you can put these two bits together (`"_tuple" << "[0].ravel()"`).  And actually I think you can simplify the code overall if only lines 135-139 are in the body of the `if`---everything else appears to be the same.